packages/samples/mvp-dapp/src/main.ts

```diff
@@ -48,12 +40,8 @@ async function bootstrap() {
   const port = kuaiEnv.config.port || 3000
   const host = kuaiEnv.config.host || '127.0.0.1'
 
-  Reflect.defineMetadata(ProviderKey.Actor, { ref: new ActorReference('resource', '/').json }, Manager)
-
   const dataSource = new NervosChainSource(kuaiEnv.config.ckbChain.rpcUrl)
-  const listener = new TipHeaderListener(dataSource)
-  const manager = new Manager(listener, dataSource)
-  manager.listen()
+  initiateResourceBindingManager(dataSource, new TipHeaderListener(dataSource))
```

Comment:
Would it be more convenient for users if this step is integrated in the project template at https://github.com/ckb-js/kuai/blob/develop/packages/core/sample-projects/src/main.ts

A similar idea was propose in my tutorial draft at https://github.com/ckb-js/kuai/issues/306

> 
> #### Chain Source
> 
> The [**Chain Source**](https://github.com/yanguoyu/kuai/blob/b988eedbb224e15f5bb1d374e8d0345b8a558dc7/packages/samples/mvp-dapp/src/chain-source.ts) module synchronizes data from CKB Node to Actor Models. It will be supported internally in the future and can be skipped in the code tour now.

Reply:
How about to initiate the resource binding in framework, and provide the `initiateResourceBindingManager` for developers to inject their own data source and listener?

Reply:
> Would it be more convenient for users if this step is integrated in the project template at https://github.com/ckb-js/kuai/blob/develop/packages/core/sample-projects/src/main.ts
> 
> A similar idea was propose in my tutorial draft at #306
> 
> > #### Chain Source
> > The [**Chain Source**](https://github.com/yanguoyu/kuai/blob/b988eedbb224e15f5bb1d374e8d0345b8a558dc7/packages/samples/mvp-dapp/src/chain-source.ts) module synchronizes data from CKB Node to Actor Models. It will be supported internally in the future and can be skipped in the code tour now.

According to this, I think these as follow could also be integrated in template, right?

```typescript
if (kuaiEnv.config.redisPort) {
    mqContainer.bind(REDIS_PORT_SYMBOL).toConstantValue(kuaiEnv.config.redisPort)
  }

  if (kuaiEnv.config.redisHost) {
    mqContainer.bind(REDIS_HOST_SYMBOL).toConstantValue(kuaiEnv.config.redisHost)
  }

  config.initializeConfig(
    config.createConfig({
      PREFIX: kuaiEnv.config.ckbChain.prefix,
      SCRIPTS: kuaiEnv.config.ckbChain.scripts || {
        ...(await getGenesisScriptsConfig(kuaiEnv.config.ckbChain.rpcUrl)),
      },
    }),
  )
```

Reply:
> > Would it be more convenient for users if this step is integrated in the project template at [`develop`/packages/core/sample-projects/src/main.ts](https://github.com/ckb-js/kuai/blob/develop/packages/core/sample-projects/src/main.ts?rgh-link-date=2023-06-13T06%3A44%3A17Z)
> > A similar idea was propose in my tutorial draft at #306
> > > #### Chain Source
> > > The [**Chain Source**](https://github.com/yanguoyu/kuai/blob/b988eedbb224e15f5bb1d374e8d0345b8a558dc7/packages/samples/mvp-dapp/src/chain-source.ts) module synchronizes data from CKB Node to Actor Models. It will be supported internally in the future and can be skipped in the code tour now.
> 
> According to this, I think these as follow could also be integrated in template, right?
> 
> ```ts
> if (kuaiEnv.config.redisPort) {
>     mqContainer.bind(REDIS_PORT_SYMBOL).toConstantValue(kuaiEnv.config.redisPort)
>   }
> 
>   if (kuaiEnv.config.redisHost) {
>     mqContainer.bind(REDIS_HOST_SYMBOL).toConstantValue(kuaiEnv.config.redisHost)
>   }
> 
>   config.initializeConfig(
>     config.createConfig({
>       PREFIX: kuaiEnv.config.ckbChain.prefix,
>       SCRIPTS: kuaiEnv.config.ckbChain.scripts || {
>         ...(await getGenesisScriptsConfig(kuaiEnv.config.ckbChain.rpcUrl)),
>       },
>     }),
>   )
> ```

Sure, it can be delivered to users by default, so they don't have to think about how to configure them.